lib/parser/index.js

```diff
@@ -520,6 +520,13 @@ Parser.prototype._insertElement = function (token, namespaceURI) {
     this.openElements.push(element);
 };
 
+Parser.prototype._insertFakeElement = function (tagName) {
+    var element = this.treeAdapter.createElement(tagName, NS.HTML, []);
+
+    this._attachElementToTree(element);
```

Comment:
As far as I remember there is only one such case: for `</br>`. But it's can be easily solved by immediately poping element out of the stack. For all other cases we need insertion and I believe we shouldn't introduce additional condition for this.